SyntheticGeometry/ProjectiveSpace/ConstancyOfFunctions.lagda.md

```diff
@@ -0,0 +1,146 @@
+Constancy of functions ℙⁿ → k
+=============================
+We show that all functions ℙⁿ → k are constant,
+assuming that all functions ℙ¹ → k are constant
+(which we did not formalize yet).
+
+```agda
+{-# OPTIONS --safe #-}
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Function
+
+import Cubical.HITs.PropositionalTruncation as PT
+open import Cubical.Data.Nat as ℕ using (ℕ; _+_)
+open import Cubical.Data.FinData
+
+open import Cubical.Algebra.CommRing
+open import Cubical.Algebra.CommRing.LocalRing
+
+open import Cubical.Relation.Nullary.Base using (¬_; yes; no)
+
+import SyntheticGeometry.SQC
+
+module SyntheticGeometry.ProjectiveSpace.ConstancyOfFunctions
+  {ℓ : Level}
+  (k : CommRing ℓ)
+  (k-local : isLocal k)
+  (k-sqc : SyntheticGeometry.SQC.sqc-over-itself k)
+  where
+
+open import SyntheticGeometry.ProjectiveSpace k k-local k-sqc
+open import SyntheticGeometry.ProjectiveSpace.StandardPoints k k-local k-sqc
+open import SyntheticGeometry.ProjectiveSpace.P0 k k-local k-sqc
+open import SyntheticGeometry.ProjectiveSpace.LineThroughPoints k k-local k-sqc
+```
+
+The case n = 0 is simple.
+
+```agda
+module n=0-Case
+  where
+
+  all-functions-constant :
+    (f : ℙ 0 → ⟨ k ⟩) →
+    2-Constant f
+  all-functions-constant f p q = cong f (isContr→isProp isContr-ℙ⁰ p q)
+```
+
+Let us now deduce the case n ≥ 1 from the case n = 1.
+
```

Comment:
Maybe this would be a good place to explain the proof? Something like:
"For a given function f:P^n -> R we want to show that the values on any pair of points are equal.
The idea is to use the fact that for *distinct* points p,q we can find a map from P¹ to P^n with p and q in its image, so the result for n=1 shows that f(p)=f(q). We will show this fact first and conclude by constructing points ..."

Reply:
Yes, look great!